apps/server/src/modules/h5p-editor/service/contentStorage.service.ts

```diff
@@ -0,0 +1,335 @@
+		await this.storageClient.delete([filePath]);
+	}
+
+	public async fileExists(contentId: string, filename: string): Promise<boolean> {
+		this.checkFilename(filename);
+
+		const filePath = this.getFilePath(contentId, filename);
+
+		return this.exists(filePath);
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public async getFileStats(contentId: string, file: string, _user: ILumiUser): Promise<IFileStats> {
+		const filePath = this.getFilePath(contentId, file);
+		const { ContentLength, LastModified } = await this.storageClient.head(filePath);
+
+		if (ContentLength === undefined || LastModified === undefined) {
+			throw new InternalServerErrorException(
+				{ ContentLength, LastModified },
+				'ContentStorage:getFileStats ContentLength or LastModified are undefined'
+			);
+		}
+
+		const fileStats: IFileStats = {
+			birthtime: LastModified,
+			size: ContentLength,
+		};
+
+		return fileStats;
+	}
+
+	public async getFileStream(
+		contentId: string,
+		file: string,
+		_user: ILumiUser,
+		rangeStart = 0,
+		rangeEnd?: number
+	): Promise<Readable> {
+		const filePath = this.getFilePath(contentId, file);
+
+		let range: string;
+		if (rangeEnd === undefined) {
+			// Open ended range
+			range = `${rangeStart}-`;
+		} else {
+			// Closed range
+			range = `${rangeStart}-${rangeEnd}`;
+		}
+
+		const fileResponse = await this.storageClient.get(filePath, range);
+		return fileResponse.data;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public async getMetadata(contentId: string, _user?: ILumiUser | undefined): Promise<IContentMetadata> {
+		const h5pContent = await this.repo.findById(contentId);
+		return h5pContent.metadata;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public async getParameters(contentId: string, _user?: ILumiUser | undefined): Promise<unknown> {
+		const h5pContent = await this.repo.findById(contentId);
+		return h5pContent.content;
+	}
+
+	public async getUsage(library: ILibraryName): Promise<{ asDependency: number; asMainLibrary: number }> {
+		const defaultUser: ILumiUser = {
+			canCreateRestricted: false,
+			canInstallRecommended: false,
+			canUpdateAndInstallLibraries: false,
+			email: '',
+			id: '',
+			name: 'getUsage',
+			type: '',
+		};
+
+		const contentIds = await this.listContent();
+		const result = await this.resolveDependecies(contentIds, defaultUser, library);
+		return result;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public getUserPermissions(_contentId: string, _user: ILumiUser): Promise<Permission[]> {
+		const permissions = [Permission.Delete, Permission.Download, Permission.Edit, Permission.Embed, Permission.View];
+
+		return Promise.resolve(permissions);
+	}
+
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	public async listContent(_user?: ILumiUser): Promise<string[]> {
+		const contentList = await this.repo.getAllContents();
```

Comment:
where can I find the authorisation for this? this seems to load all contents from the entire database...

Reply:
why was this resolved? cant see a change or response

Reply:
it was not resolved seems as an internal method from Lumi interface, but we don't use it yet, may it will be finished in the next feature Tickets